src/experimental/tools/mediaCapabilitiesProber/api/index.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import log from "../log";
+import {
+  IDisplayConfiguration,
+  IDRMConfiguration,
+  IMediaConfiguration,
+} from "../types";
+import probeMediaConfiguration, { IBrowserAPIS } from "./probeMediaConfiguration";
+
+/**
+ * A set of API to probe media capabilites.
+ * Each API allow to evalute a specific feature (HDCP support, decoding infos, etc)
+ * and relies on different browser API to probe capabilites.
+ */
+const mediaCapabilitiesProber = {
+
+  /**
+   * Set logger level
+   * @param {string} level
+   */
+  set LogLevel(level: string) {
+    log.setLevel(level);
+  },
+
+  /**
+   * Get logger level
+   * @returns {string}
+   */
+  get LogLevel() : string {
+    return log.getLevel();
+  },
+
+  /**
+   * Get HDCP status. Evaluates if current equipement support given
+   * HDCP revision.
+   * @param {string}
+   * @returns {Promise}
+   */
+  getStatusForHDCP(hdcp: string) : Promise<string> {
+    if (!hdcp) {
+      return Promise.reject("MediaCapabilitiesProbers >>> Bad Arguments: " +
+        "No HDCP Policy specified.");
+    }
+    const config = {
+      hdcp,
+    };
+    const browserAPIS: IBrowserAPIS[] = [
+      "isTypeSupportedWithFeatures",
+      "getStatusForPolicy",
+    ];
+    return probeMediaConfiguration(config, browserAPIS).then((result) => {
+      switch (result) {
+        case 0:
+          return "NotSupported";
+        case 1:
+          return "Unknown";
+        case 2:
+          return "Supported";
+        default:
+          return "NotSupported";
+      }
+    });
+  },
+
+  /**
+   * Get decoding capabilities from a given video and/or audio
+   * configuration.
+   * @param {Object} mediaConfig
+   * @returns {Promise}
+   */
+  getDecodingCapabilities(
+    mediaConfig: IMediaConfiguration
+  ) : Promise<string> {
+    const config = {
+      type: mediaConfig.type,
+      video: mediaConfig.video,
+      audio: mediaConfig.audio,
+    };
+    const browserAPIS: IBrowserAPIS[] = [
+      "isTypeSupported",
+      "isTypeSupportedWithFeatures",
+      "decodingInfos",
+    ];
+    return probeMediaConfiguration(config, browserAPIS).then((result) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 66. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=236&issues=AWRg5q1zI66nRh8Y2ome&open=AWRg5q1zI66nRh8Y2ome)

[ISSUE_KEY]: AWRg5q1zI66nRh8Y2ome